### PR TITLE
Add CodiMD OCA

### PR DESCRIPTION
**Pull request by zuedev**
Adds [CodiMD](https://github.com/hackmdio/codimd) one-click app. Using MariaDB as DB driver. End instructions aren't really needed, but I can expound them if absolutely needed.

Thank you for the awesome platform! Looking forward to contributing more tested apps to the repo. 😉 ❤️ 

### ☑️ Self Check before Merge

- [x] I have tested the template using the method described in README.md thoroughly
- [x] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [x] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [x] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [x] Icon is added as a png file to the logos directory.

### CODE REVIEW

File: public/v4/apps/codimd.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        restart: always
+        volumes:
+            - $$cap_appname-config:/config
+        caproverExtra:
+            containerHttpPort: '3000'
+caproverOneClickApp:
+    variables:
+        - label: Database Password
+          description: Password to connect to the database
+          defaultValue: daTaBaSePaSsWorD
```
Comment (githubsaturn) [inline public/v4/apps/codimd.yml:35]: You can use `$$cap_gen_random_hex(16)`

File: public/v4/apps/codimd.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          id: $$cap_DB_PASS
+    instructions:
+        start: Realtime collaborative markdown notes on all platforms
```
Comment (githubsaturn) [inline public/v4/apps/codimd.yml:38]: Please add more details here. Perhaps a link to github or the official website.

File: public/v4/apps/codimd.yml (hunk 1 above)
Comment (zuedev) [reply public/v4/apps/codimd.yml]: Done in 3adeac0

File: public/v4/apps/codimd.yml (hunk 2 above)
Comment (zuedev) [reply public/v4/apps/codimd.yml]: Answered in 3adeac0, is that better? 😄

**Comment by githubsaturn** [comment]
Thanks!

**Pull request merged**